src/parser.js

```diff
@@ -34,38 +46,113 @@ export function containIncludeCommand(commands = []) {
         return reg.test(command.trim());
     })
 }
+/** Parse the command label and return key-value object
+ * @example
+ *      [import,title:"<thetitle>",label:"<thelabel>"](path/to/file.ext)
+ * @param {string} label
+ * @return {Array}
+ */
+export function parseVariablesFromLabel(label) {
+    var keyvals = {
+        "title":undefined,
+        "id":undefined,
+        "marker":undefined
+    };
+    for(var key in keyvals) {
+        var keyReg=key;
+        if(key=="marker") {
+            keyReg="import|include";
+        }
+        const regStr = "\^.*,?\\s*("+keyReg+")\\s*:\\s*[\"']([^'\"]*)[\"'],?.*\$";
+        const reg = new RegExp(regStr);
+        const res = label.match(reg);
+        if(res) {
+            keyvals[key]=res[2];
+        }
+    }
+    return keyvals;
+}
 
 /**
  * generate code with options
  * @param {string} lang
  * @param {string} filePath
  * @param {string} originalPath
  * @param {string} label
+ * @return {string}
  */
 export function embedCode({lang, filePath, originalPath, label}) {
     const code = fs.readFileSync(filePath, "utf-8");
     const fileName = path.basename(filePath);
+    const keyValueObject = parseVariablesFromLabel(label);
+    var content = code;
+    // Slice content via line numbers.
     if (hasSliceRange(label)) {
         const [start, end] = getSliceRange(label);
-        const content = sliceCode(code, start, end);
-        return generateEmbedCode(lang, fileName, originalPath, content);
-    } else if (hasMarker(label)) {
-        const marker = getMarkerName(label);
-        const content = removeMarkers(markersSliceCode(code, marker));
-        return generateEmbedCode(lang, fileName, originalPath, content);
-    } else {
-        return generateEmbedCode(lang, fileName, originalPath, code);
+        content = sliceCode(code, start, end);
+    }
+    // Slice content via markers.
+    else if (hasMarker(keyValueObject)) {
+        const marker = getMarker(keyValueObject);
+        content = removeMarkers(markerSliceCode(code, marker));
     }
+    return generateEmbedCode(keyValueObject, lang, fileName, originalPath, content);
 }
 
-export function generateEmbedCode(lang, fileName, originalPath, content) {
-    return `> <a name="${fileName}" href="${originalPath}">${fileName}</a>
+/**
+ * generate code from options
+ * @param keyValueObject
```

Comment:
Please put `{Object}` type to `keyValueObject`.
`@param {Object} keyValueObject` means that `keyValueObject` is object.